### PR TITLE
Clean up BSPs

### PR DESCRIPTION
**Pull request by thejpster**
Gets them all to a 0.1.0 release.

You might want to do https://github.com/rp-rs/rp-hal/issues/244 instead (but at least they're in a state where they can be separated out fairly cleanly).

### CODE REVIEW

**Review by 9names — APPROVED** [review]
Looks good, happy to merge this even if we decided to go through with the #244 split.

If we're prefixing all boards with a vendor name should we also rename pico to raspberry-pi-pico to be consistent?
Or is it obvious enough that Pico is from Raspberry Pi and the prefix adds no value?

**Comment by anall** [comment]
If we're publishing it, `pico` seems like something we should avoid claiming? -- too generic of a name.

**Comment by 9names** [comment]
I hadn't considered that, but also since it's generic and short it is also already taken.
https://crates.io/crates/pico

**Comment by thejpster** [comment]
I did wonder if I should rename pico. We probably should. `raspberrypi-pico` or `raspberry-pi-pico`?

**Comment by thejpster** [comment]
Oh one more thing I thought about but haven't yet done - should the crate owners all be reset to "The rp-rs Developers"?

**Comment by 9names** [comment]
I like `raspberrypi-pico` more, don't really need another dash to disambiguate.

> should the crate owners all be reset to "The rp-rs Developers"

Owners in which respect? If you mean the authors field in Cargo.toml that's fine by me, would be good to get quorum before doing that though.

**Comment by jannic** [comment]
What about `rp-pico`?

**Comment by thejpster** [comment]
> What about `rp-pico`?

OK, that does have a certain consistency to it.

**Comment by thejpster** [comment]
> Oh one more thing I thought about but haven't yet done - should the crate owners all be reset to "The rp-rs Developers"?

@ithinuel
@hmvp 
@eolder
@onnenon 
@anall 

Are you happy I reset your BSP crates to have an author of "The rp-rs Developers" in the relevant `Cargo.toml` files?

File: boards/rp-pico/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "pico"
+name = "rp-pico"
 version = "0.1.0"
```
Comment (jannic) [inline boards/rp-pico/Cargo.toml]: Version needs to be higher for `rp-pico`, as there is already an older BSP with version `0.1.3` with that name.
So bump it to `0.2.0`, I guess?

Comment (thejpster) [reply boards/rp-pico/Cargo.toml]: https://github.com/rp-rs/rp-hal/pull/245/commits/6271e8670fe61cafb48f636ceaf5e001949a1442

**Comment by dylanede** [comment]
I noticed that a few of the BSPs have dependencies they don't use. For example adafruit-macropad depends on `cortex-m` even though it does not import anything from it. Is that intentional? If not maybe cleaning that up should be included in this PR.

**Comment by thejpster** [comment]
This PR has been merged but I'd be happy to see one that trims dependencies down the minimum set required.

**Pull request merged**